src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -183,7 +183,14 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
             catch (Exception ex)
             {
                 // profiled app will continue working as expected without this method
-                Log.ErrorException($"Error resolving {DiagnosticSourceTypeName}.{nameof(BeforeAction)}(...)", ex);
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: DiagnosticSourceTypeName,
+                    methodName: nameof(BeforeAction),
+                    instanceType: diagnosticSource?.GetType().AssemblyQualifiedName);
```

Comment:
We're instrumenting a static method, so there's no instance in this case. `diagnosticSource` is the first argument.

Reply:
I was wondering, because this is an extension method, and idiomatically the `this` parameter is often treated like an instance... how should we handle that?
Because the information IS valuable, as in this case, there are 3 types that it could be.

Reply:
I don't think that the fact that a method is called using extension method syntax is relevant. That's just a language feature that doesn't exist at the IL level.

We're not intercepting this:
```csharp
diagnosticSource.BeforeAction(actionDescriptor,
                              httpContext,
                              routeData);
```

We're intercepting this:
```csharp
MvcCoreDiagnosticSourceExtensions.BeforeAction(diagnosticSource,
                                               actionDescriptor,
                                               httpContext,
                                               routeData);
```

I think the run time type of all the arguments are equally important. We should either log all of them all or none.

Reply:
I'm not logging any of the parameters because it would have been a lot to pass around IMO. I'm okay with passing null if you feel strongly about it. I felt it might be reasonably valuable information.

Reply:
My thought was: "What is enough information for us to figure out what's going on?"